modules/nextflow/src/main/groovy/nextflow/executor/LsfExecutor.groovy

```diff
@@ -35,6 +35,37 @@ class LsfExecutor extends AbstractGridExecutor {
 
     private static char BLANK = ' ' as char
 
+    /**
+    * parse the memory units from $LSF_ENVDIR/lsf.conf, 
+    * which should be accessible via all nodes. Otherwise, default is KB
+    * -- see https://www.ibm.com/support/knowledgecenter/en/SSETD4_9.1.3/lsf_config_ref/lsf.conf.lsf_unit_for_limits.5.html
+    */
+    def getLSFmemoryUnits() {
+        // check environment variable exists
+        def envDir = System.getenv("LSF_ENVDIR") 
+        if ( !envDir ) return null
+        def envFile = new File(envDir, "lsf.conf")
+        if ( !envFile.exists() ) return null
+        for (def line : envFile.readLines() ){
+            if ( line.startsWith("LSF_UNIT_FOR_LIMITS=") ){
+                def memoryUnits = line.split("LSF_UNIT_FOR_LIMITS=")[1]
+            }
+        }
+        return(memoryUnits)
+    }
```

Comment:
I expect this to not compile because `memoryUnits` if defined in the for inner scope

Reply:
uff